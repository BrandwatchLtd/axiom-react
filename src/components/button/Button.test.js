import React from 'react';
import renderer from 'react-test-renderer';
import { Button, ButtonIcon } from 'bw-axiom';

function getComponent(props = {}, children = 'Lorem Ipsum') {
  return renderer.create(
    <Button { ...props }>{ children }</Button>
  );
}

describe('Button', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with full', () => {
    [true, 'small', 'medium', 'large'].forEach((full) => {
      it(full, () => {
        const component = getComponent({ full });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with size', () => {
    ['small', 'medium', 'large'].forEach((size) => {
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with style', () => {
    ['primary', 'secondary', 'tertiary'].forEach((style) => {
      it(style, () => {
        const component = getComponent({ style });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with circular', () => {
    ['small', 'medium', 'large', 'huge'].forEach((circular) => {
      it(circular, () => {
        const component = getComponent({ circular });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('with ButtonIcon', () => {
    it('adds space to start if first child', () => {
      const component = getComponent({}, [
        <ButtonIcon name="twitter" />,
        'Lorem ipsum',
      ]);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('adds space to end if last child', () => {
      const component = getComponent({}, [
        'Lorem ipsum',
        <ButtonIcon name="twitter" />,
      ]);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('adds space all around if middle children', () => {
      const component = getComponent({}, [
        'Lorem ipsum',
        <ButtonIcon name="twitter" />,
        'Lorem ipsum',
      ]);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
