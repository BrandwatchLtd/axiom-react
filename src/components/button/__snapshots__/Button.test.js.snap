// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with circular huge 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--circular-huge"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with circular large 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--circular-large"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with circular medium 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--circular-medium"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with circular small 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--circular-small"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with defaultProps 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with full large 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--full--large"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with full medium 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--full--medium"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with full small 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--full--small"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with full true 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--full"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with size large 1`] = `
<button
  className="ax-button ax-button--large ax-button--primary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with size medium 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with size small 1`] = `
<button
  className="ax-button ax-button--small ax-button--primary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with stadium small 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary ax-button--stadium-small"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with style primary 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with style quaternary 1`] = `
<button
  className="ax-button ax-button--medium ax-button--quaternary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with style secondary 1`] = `
<button
  className="ax-button ax-button--medium ax-button--secondary"
>
  Lorem Ipsum
</button>
`;

exports[`Button renders with style tertiary 1`] = `
<button
  className="ax-button ax-button--medium ax-button--tertiary"
>
  Lorem Ipsum
</button>
`;

exports[`Button with ButtonIcon adds space all around if middle children 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary"
>
  Lorem ipsum
  <span
    className="ax-button__icon"
  >
    <svg
      className="ax-icon"
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    >
      <path
        d="M1,12.087 C1.225,12.113 1.454,12.127 1.685,12.127 C3.031,12.127 4.27,11.667 5.252,10.897 C3.995,10.874 2.935,10.044 2.569,8.902 C2.745,8.936 2.925,8.954 3.11,8.954 C3.372,8.954 3.716,8.919 3.957,8.853 C2.643,8.589 1.743,7.429 1.743,6.037 C1.743,6.216 2.483,6.345 2.954,6.36 C2.183,5.845 1.631,4.966 1.631,3.969 C1.631,3.443 1.75,2.949 1.997,2.525 C3.414,4.263 5.519,5.407 7.906,5.526 C7.857,5.316 7.826,5.097 7.826,4.872 C7.826,3.286 9.11,2 10.696,2 C11.522,2 12.267,2.348 12.791,2.907 C13.445,2.778 14.059,2.539 14.614,2.209 C14.399,2.88 13.944,3.443 13.351,3.799 C13.932,3.729 14.485,3.575 15,3.346 C14.615,3.922 14.128,4.428 13.567,4.833 C13.573,4.956 13.575,5.08 13.575,5.205 C13.575,9 10.686,13.377 5.403,13.377 C3.781,13.377 2.271,12.901 1,12.087 Z"
      />
    </svg>
  </span>
  Lorem ipsum
</button>
`;

exports[`Button with ButtonIcon adds space to end if last child 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary"
>
  Lorem ipsum
  <span
    className="ax-button__icon ax-button__icon--end"
  >
    <svg
      className="ax-icon"
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    >
      <path
        d="M1,12.087 C1.225,12.113 1.454,12.127 1.685,12.127 C3.031,12.127 4.27,11.667 5.252,10.897 C3.995,10.874 2.935,10.044 2.569,8.902 C2.745,8.936 2.925,8.954 3.11,8.954 C3.372,8.954 3.716,8.919 3.957,8.853 C2.643,8.589 1.743,7.429 1.743,6.037 C1.743,6.216 2.483,6.345 2.954,6.36 C2.183,5.845 1.631,4.966 1.631,3.969 C1.631,3.443 1.75,2.949 1.997,2.525 C3.414,4.263 5.519,5.407 7.906,5.526 C7.857,5.316 7.826,5.097 7.826,4.872 C7.826,3.286 9.11,2 10.696,2 C11.522,2 12.267,2.348 12.791,2.907 C13.445,2.778 14.059,2.539 14.614,2.209 C14.399,2.88 13.944,3.443 13.351,3.799 C13.932,3.729 14.485,3.575 15,3.346 C14.615,3.922 14.128,4.428 13.567,4.833 C13.573,4.956 13.575,5.08 13.575,5.205 C13.575,9 10.686,13.377 5.403,13.377 C3.781,13.377 2.271,12.901 1,12.087 Z"
      />
    </svg>
  </span>
</button>
`;

exports[`Button with ButtonIcon adds space to start if first child 1`] = `
<button
  className="ax-button ax-button--medium ax-button--primary"
>
  <span
    className="ax-button__icon ax-button__icon--start"
  >
    <svg
      className="ax-icon"
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    >
      <path
        d="M1,12.087 C1.225,12.113 1.454,12.127 1.685,12.127 C3.031,12.127 4.27,11.667 5.252,10.897 C3.995,10.874 2.935,10.044 2.569,8.902 C2.745,8.936 2.925,8.954 3.11,8.954 C3.372,8.954 3.716,8.919 3.957,8.853 C2.643,8.589 1.743,7.429 1.743,6.037 C1.743,6.216 2.483,6.345 2.954,6.36 C2.183,5.845 1.631,4.966 1.631,3.969 C1.631,3.443 1.75,2.949 1.997,2.525 C3.414,4.263 5.519,5.407 7.906,5.526 C7.857,5.316 7.826,5.097 7.826,4.872 C7.826,3.286 9.11,2 10.696,2 C11.522,2 12.267,2.348 12.791,2.907 C13.445,2.778 14.059,2.539 14.614,2.209 C14.399,2.88 13.944,3.443 13.351,3.799 C13.932,3.729 14.485,3.575 15,3.346 C14.615,3.922 14.128,4.428 13.567,4.833 C13.573,4.956 13.575,5.08 13.575,5.205 C13.575,9 10.686,13.377 5.403,13.377 C3.781,13.377 2.271,12.901 1,12.087 Z"
      />
    </svg>
  </span>
  Lorem ipsum
</button>
`;
